packages/amplify-cli-core/src/customPoliciesUtils.ts

```diff
@@ -11,7 +11,7 @@ export type CustomIAMPolicies = CustomIAMPolicy[];
 export type CustomIAMPolicy = {
   Action: string[];
   Effect?: string;
-  Resource: string[];
+  Resource: (string | object)[];
```

Comment:
Can you change this to `Resource: (string | $TSObject)[];`?

You will need to import the `$TSObject` type from `amplify-cli-core/src/index.ts`

Reply:
Implemented in latest commit https://github.com/aws-amplify/amplify-cli/pull/9076/commits/343b339ebfb58c41d01592a0aa25d1508370cd0a